velodrome/transform/first_comment.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/test-infra/velodrome/sql"
+)
+
+type IssueCreation struct {
+	Author      string
+	CreatedAt   time.Time
+	Pushed      bool
+	PullRequest bool
+}
+
+// FirstComment plugin: Count number
+type FirstComment struct {
+	DB     InfluxDatabase
+	last   time.Time
+	issues map[int]IssueCreation
+}
+
+// NewFirstCommentPlugin initializes the plugin. Requires an
+// InfluxDatabase to push the metric
+func NewFirstCommentPlugin(DB InfluxDatabase) *FirstComment {
+	last, err := DB.GetLastMeasurement("first_comment")
+	if err != nil {
+		glog.Fatal("Failed to create FirstComment plugin: ", err)
+	}
+	return &FirstComment{
+		DB:     DB,
+		last:   last,
+		issues: make(map[int]IssueCreation),
+	}
+}
+
+// ReceiveIssue is not used for this metric
+func (m *FirstComment) ReceiveIssue(issue sql.Issue) error {
+	if !issue.IsPR && issue.FindLabels(regexp.MustCompile(`kind/flakes`)) == nil {
+		return nil
+	}
+	if _, ok := m.issues[issue.ID]; ok {
+		return nil
+	}
+
+	m.issues[issue.ID] = IssueCreation{
+		Author:      issue.User,
+		CreatedAt:   issue.IssueCreatedAt,
+		PullRequest: issue.IsPR,
+		Pushed:      false,
+	}
+
+	return nil
+}
+
+func (m *FirstComment) Process(issueID int, author string, date time.Time) error {
```

Comment:
I'm a little confused by the logic here. Can you make a comment describing exactly what this is trying to do?